File: src/main/java/com/nwmqpa/graphcall/nodes/ParameterField.java
package com.nwmqpa.graphcall.nodes;

import com.nwmqpa.graphcall.interfaces.ICompilable;

public class ParameterField implements ICompilable {

    private final String parameter;
    private final Object value;

    public ParameterField(String parameter, Object value) {
        this.parameter = parameter;
        this.value = value;
    }

    @Override
    public String compile() {
        if (this.value instanceof Integer)
            return String.format("%s: %d", this.parameter, this.value);
        return String.format("%s: \\\"%s\\\"", this.parameter, this.value);
    }
}

File: python/graphcall/nodes/parameter_field.py
"""Module defining the parameter field of a Node."""
from graphcall.interfaces.compilable import Compilable


class ParameterField(Compilable):
    """Class defining the parameter field."""

    def __init__(self, parameter: str, value) -> None:
        """Initialize the object."""
        self.parameter = parameter
        self.value = value

    def compile(self):
        """Compile the parameter field."""
        if isinstance(self.value, str):
            return "{0}: \"{1}\"".format(self.parameter, self.value)
        if isinstance(self.value, bool):
            return "{0}: {1}".format(self.parameter, str(self.value).lower())
        return "{0}: {1}".format(self.parameter, str(self.value))
